_posts/2023-04-18-searchable-snapshots.md

```diff
@@ -0,0 +1,86 @@
+has_science_table: true
+---
+
+The OpenSearch Project is excited to announce the general availability of searchable snapshots in OpenSearch. With searchable snapshots, you can search indexes that are stored as snapshots within remote repositories in real time without the need to restore the index data to local storage ahead of time. The relevant index data is retrieved on demand with the search request. Searchable snapshots cache data on local storage to improve the performance. In this blog post, we will cover the design and implementation of searchable snapshots, their performance, and planned future enhancements. 
+
+## Using a snapshot for search
+
+Before searchable snapshots, searching data within a snapshot required restoring the entire snapshot to the OpenSearch cluster node. This involved expanding your infrastructure, which is costly and time-consuming. 
+
+The searchable snapshots feature caches frequently used data segments in cluster nodes, using the Least Recently Used (LRU) caching strategy to remove the least used data segments from the cluster nodes in order to make space for frequently used ones. The data segments downloaded from snapshots on block storage reside alongside the general indexes of the cluster nodes. The local storage of the node is used for caching the snapshot data. Any subsequent searches on the cached data perform better because they are served from the node's local storage, or cache, without the need to redownload the segment data from the snapshot. Therefore, you can search snapshots stored in lower-cost storage, like Amazon S3, with acceptable performance. 
+
+The following image shows the creation of an index using a remote storage option, such as Amazon S3.
+
+![Searchable Snapshots](/assets/media/blog-images/2023-04-27-searchable-snapshots/searchable-snapshot.png){: .img-fluid}
+
+## Enabling searchable snapshots in OpenSearch
+
+To enable searchable snapshots, reference the [instructions](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/searchable_snapshot/) in the OpenSearch documentation.
+
+## Performance
+
+Comparing the performance of searchable snapshot queries and local index queries on the `stackoverflow` dataset, the initial run of searchable snapshots took approximately twice as long as the local index. However, subsequent queries showed similar performance for both searchable snapshots and local index because searchable snapshots used cached data. 
+
+The following table presents the time in seconds taken to scan approximately 36 million documents in the [`stackoverflow`](https://www.kaggle.com/datasets/stackoverflow/stackoverflow) dataset using the Scroll API in a single-threaded test runner.
```

Comment:
1. why not publish the data for non-scroll standard search request based queries ?
2. Also do we need to talk about setup like a) host types used, b) number of hosts, c) number of shards per host, etc ? Same for benchmark section

Reply:
> why not publish the data for non-scroll standard search request based queries ?

>> The benchmark section below shows data for the nyc_taxis workload

Was not clear from the benchmark section below. Thanks for adding the note to update it.

> Also do we need to talk about setup like a) host types used, b) number of hosts, c) number of shards per host, etc ? Same for benchmark section

>> I think the relative performance to the local index is what is important here. I'm not sure that adding more detail will be super helpful. What do you think?

First scan time can vary depending on the local disk type used in the setup (SSD vs HDD vs EBS) and the network bandwidth. Relative performance is fine but it may change with different setup like 1 active shard vs 10 active shards per node.